magento/magento2
# Fixed tier price issue on configurable product

### Description (*)

The tier price for the configurable product isn’t deleted when a simple product is converted into a configurable product. The Actual price of a product is removed from the database but the tier price  remains there in `catalog_product_entity_tier_price`. 
### Related Pull Requests


### Fixed Issues (if relevant)


1. Fixes  #38390 

### Manual testing scenarios (*)

1. Create a simple product with a tier price.
2. Edit that simple product and convert it into a configurable product by creating some variants with the price greater than the tier price. Variants should not have any tier price.
3. Observe in the database that table `catalog_product_entity_tier_price` still has the entry for the particular product with the tier price.
4. Visit the particular product on a listing page and product page and observe the price.

### Additional Information

File:magento/module-catalog/Model/Product/Attribute/Backend/TierPrice/UpdateHandler.php
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/UpdateHandler.php#L83-L92
The problem lies in the fact that the tier price handler is called every time we update the product, but it fails to remove the tier price value from the database when we convert a simple product into a configurable product.

### Solution
To resolve this issue, we need to add a condition. For deleting the tier price from the database, set the $priceRows variable as an empty array. Afterward, in this file, this triggers the condition on line  92 to become true, and a delete action is performed. Before executing this action, it's crucial to check if the product type is configurable. Also, consider whether the simple product has a tier price. After that, a condition is added: If the product type is configurable and the simple product has a tier price, the $priceRows variable is set to an empty array. Following this adjustment, the tier price of the configurable product is removed from the database table.


### Contribution checklist (*)
